### DISCUSSION TITLE
`${hydra.job.override_dirname}` with `/` override creates redundant directories

### DISCUSSION THREAD

**Discussion by Aceticia** [facebookresearch/hydra#2409] [Q&A] [answered]
I'm using the override_dirname as the folder to store checkpoints and logs. This does store the logs, but since my configs contain overrides with `/`, hydra ended up not treating them as part of the subdir but as a directory under the subdir, creating one redundant level of folder each time I override something. This is a pretty small inconvenience, and it would probably be very difficult to fix (how to differentiate where user actually want a subdirectory but not override?), but I imagine it's not the intended behavior?

For more details, here is how I'm overriding some configs. If there is an alternative where I can just not use `/` overrides I guess that would solve the issue too. 

```
models
- default.yaml
- temporal_nets
    - net1.yaml
    - net2.yaml
```

In `net1.yaml` and `net2.yaml`, there are configurations for neural networks including `__target__` instantiations and their hyperparameters. In `default.yaml` I simply have a default list of using `net1.yaml`. When I do need to run an experiment with net2, I just override with `models/temporal_nets=net2`. What are alternative ways to organize the configs, so that I can override the choice of `temporal_nets`? Any suggestions would be appreciated. Thank you.

**Comment by Jasha10** [comment, marked as answer]
> What are alternative ways to organize the configs, so that I can override the choice of temporal_nets?

There are certainly ways to avoid using a slash `/` in the overrides, e.g. doing `models_temporal_nets=net2` instead of `models/temporal_nets=net2`. That being said, using a slash in the override is convenient (it lets you organize your yaml configs in a tree of directories and access those configs in a straightforward way).

Instead of changing the overrides that you use, I'd recommend using an [OmegaConf resolver](https://omegaconf.readthedocs.io/en/2.2_branch/custom_resolvers.html) to sanitize the override_dirname. This would mean replacing `${hydra:job.override_dirname}` with `${sanitize_override_dirname:${hydra:job.override_dirname}}`, where `sanitize_override_dirname` is a function you register using `OmegaConf.register_new_resolver`. This would enable you to e.g. replace the slashes with underscores (or other symbols) in the override dirname.
